src/main/webapp/script.js

```diff
@@ -6,7 +6,13 @@
  */
 
 function initBody() {
-    checkPermissions();
+  setCredentialsServlet();
+  checkPermissions();
+}
+
+function setCredentialsServlet() {
+  fetch('/get-credentials?projID='+config.projectID+'&bucket='+config.bucketName+"&object="+config.objectName)
```

Comment:
You should encode the values using encodeURIComponent

Also, Consider using a helper function to build your URLs. I quickly put this together that you can use as a starting point:
```
function buildURL(url, params) {
    var encodedParams = new URLSearchParams(params);
    return `${url}/${encodedParams.toString()}`;
}
```
You can use it like this:
```
buildURL("my-endpoint", {param1:"foo", param2:"bar"})
-> "my-endpoint/param1=foo&param2=bar"
```

Reply:
That's a lot easier and more readable - I didn't know it could be done that way! Thank you :)